client/dashboard/me/tax-details/user-tax-form.tsx

```diff
@@ -0,0 +1,217 @@
+							rel="noreferrer"
+							onClick={ () => {
+								recordTracksEvent( 'calypso_vat_details_support_click' );
+							} }
+						/>
+					),
+				},
+			}
+		);
+
+	return (
+		<InputControl
+			__next40pxDefaultSize
+			disabled={ isDisabled || ( isVatAlreadySet && ! canUserEdit ) }
+			help={ isVatAlreadySet && vatIdHelp }
+			label={ label }
+			onChange={ ( value ) => onChange( { [ id ]: value } ) }
+			prefix={ country && <InputControlPrefixWrapper>{ country }</InputControlPrefixWrapper> }
+			value={ getValue( { item: data } ) || '' }
+		/>
+	);
+}
+
+function VatInputControl( { data, field, onChange }: UserTaxFormControlProps ) {
+	const { getValue, id, label, isDisabled } = field;
+
+	return (
+		<InputControl
+			__next40pxDefaultSize
+			disabled={ isDisabled }
+			label={ label }
+			onChange={ ( value ) => onChange( { [ id ]: value } ) }
+			value={ getValue( { item: data } ) || '' }
+		/>
+	);
+}
+
+export default function UserTaxForm() {
+	const { recordTracksEvent } = useAnalytics();
+
+	const [ localData, setLocalData ] = useState< Partial< UserTaxFormData > >( {} );
+
+	const {
+		isLoading,
+		isUpdateSuccessful,
+		isUpdating,
+		setUserTaxDetails,
+		userTaxDetails,
+		updateError,
+	} = useUserTaxDetails();
+	const countryCodes = useDataFormCountryCodes();
+
+	const formData = useMemo( () => {
+		const serverData = {
+			country: userTaxDetails.country ?? '',
+			id: userTaxDetails.id ?? '',
+			name: userTaxDetails.name ?? '',
+			address: userTaxDetails.address ?? '',
+		};
+		return {
+			...serverData,
+			...localData,
+		};
+	}, [
+		localData,
+		userTaxDetails.address,
+		userTaxDetails.country,
+		userTaxDetails.id,
+		userTaxDetails.name,
+	] );
+
+	const { data: geoData } = useGeoLocationQuery();
+	const taxName = useTaxName( formData.country ?? geoData?.country_short ?? 'GB' );
+
+	useDisplayUserTaxNotices( { error: updateError, success: isUpdateSuccessful, taxName } );
+	useRecordUserTaxEvents( { updateError, isUpdateSuccessful } );
+
+	const isVatAlreadySet = !! userTaxDetails.id;
+	const isDisabled = isLoading || isUpdating;
+	const canUserEdit = userTaxDetails.can_user_edit ?? false;
+
+	const fields: UserTaxField[] = [
+		{
+			Edit: VatSelectControl,
+			elements: countryCodes,
+			id: 'country',
+			isDisabled,
+			isVatAlreadySet,
+			canUserEdit,
+			label: __( 'Country' ),
+		},
+		{
+			Edit: VatIdControl,
+			id: 'id',
+			isDisabled,
+			isVatAlreadySet,
+			canUserEdit,
+			label: __( 'VAT ID' ),
+			taxName,
+		},
+		{
+			Edit: VatInputControl,
+			id: 'name',
+			isDisabled,
+			label: __( 'Name' ),
+			type: 'text',
+		},
+		{
+			Edit: VatInputControl,
+			id: 'address',
+			isDisabled,
+			label: __( 'Address' ),
+			type: 'text',
+		},
+	];
+
+	const form = {
+		type: 'regular' as const,
+		fields: [ 'country', 'id', 'name', 'address' ],
+	};
+
+	const onSubmit = ( e: React.FormEvent ) => {
+		e.preventDefault();
+		recordTracksEvent( 'calypso_vat_details_update' );
```

Comment:
Yes nice, recording the tracks events is a lot clearer when done in imperative code like this.

Just a heads up that for analytics in the HD we want to use this prefix


```suggestion
		recordTracksEvent( 'calypso_dashboard_vat_details_update' );
```